mteb/abstasks/AbsTaskClassification.py

```diff
@@ -162,21 +159,36 @@ def _evaluate_subset(
         avg_scores["scores_per_experiment"] = scores
         return avg_scores
 
-    def _undersample_data(self, X, y, samples_per_label: int, idxs=None):
-        """Undersample data to have samples_per_label samples of each label"""
-        X_sampled = []
-        y_sampled = []
+    def _undersample_data(
+        self, dataset: Dataset, samples_per_label: int, idxs=None
+    ) -> tuple[Dataset, list[int]]:
+        """Undersample data to have `samples_per_label` samples of each label.
+
+        Args:
+            dataset: Hugging Face `datasets.Dataset` containing "text" and "label".
+            samples_per_label: Number of samples per label to retain.
+            idxs: Optional indices to shuffle and sample from.
+
+        Returns:
+            A new Dataset containing undersampled examples.
+            The shuffled indices used for sampling.
+        """
         if idxs is None:
-            idxs = np.arange(len(y))
+            idxs = list(range(len(dataset)))
+
         rng_state = np.random.default_rng(self.seed)
         rng_state.shuffle(idxs)
```

Comment:
```suggestion
         self.rng_state.shuffle(idxs)
```
test and believe they should be eq.

Reply:
On the first experiment they're equal and on others they're different